pyk/src/tests/integration/ktool/test_fuzz.py

```diff
@@ -0,0 +1,129 @@
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING
+
+import pytest
+
+from pyk.kast.inner import KSort
+from pyk.kore.parser import KoreParser
+from pyk.kore.prelude import inj, top_cell_initializer
+from pyk.kore.syntax import DV, App, Assoc, EVar, SortApp, String
+from pyk.ktool.kfuzz import fuzz, kintegers
+from pyk.ktool.kprint import _kast
+from pyk.testing import KompiledTest
+
+from ..utils import K_FILES, TEST_DATA_DIR
+
+if TYPE_CHECKING:
+    from collections.abc import Callable
+    from pathlib import Path
+    from typing import Final
+
+    from hypothesis.strategies import SearchStrategy
+
+    from pyk.kore.syntax import Pattern
+
+_LOGGER: Final = logging.getLogger(__name__)
+
+FUZZ_FILES: Path = TEST_DATA_DIR / 'fuzzing'
+
+VAR_X = EVar(name='VarX', sort=SortApp('SortInt'))
+VAR_Y = EVar(name='VarY', sort=SortApp('SortInt'))
+
+
+class TestImpFuzz(KompiledTest):
+    KOMPILE_MAIN_FILE = K_FILES / 'imp.k'
+    KOMPILE_BACKEND = 'llvm'
+
+    @staticmethod
+    def substs() -> dict[EVar, SearchStrategy[Pattern]]:
+        return {VAR_X: kintegers(with_inj=KSort('AExp')), VAR_Y: kintegers(with_inj=KSort('AExp'))}
+
+    @staticmethod
+    def check() -> Callable[[Pattern], None]:
```

Comment:
You can eliminate the inner function:

```python3
def check(pattern: Pattern) -> None:
```

then

```python3
fuzz(definition_dir, init_pattern, self.substs(), self.check)
```

Reply:
I couldn't quite remove it but I got rid of the ugly looking lambda.

Reply:
Makes sense.